src/MigrationActions.sol

```diff
@@ -0,0 +1,144 @@
+    }
+
+    /**
+     * @notice Migrate `assetsIn` of `dai` to `snst`.
+     * @param  receiver  The receiver of the `snst`.
+     * @param  assetsIn  The amount of the `dai` to migrate.
+     * @return sharesOut The amount of `snst` shares received.
+     */
+    function migrateDAIToSNST(address receiver, uint256 assetsIn) external returns (uint256 sharesOut) {
+        dai.transferFrom(msg.sender, address(this), assetsIn);
+        _migrateDAIToNST(address(this), assetsIn);
+        sharesOut = snst.deposit(assetsIn, receiver);
+    }
+
+    /**
+     * @notice Migrate `assetsIn` of `sdai` to `nst`.
+     * @param  receiver The receiver of the `nst`.
+     * @param  assetsIn The amount of the `sdai` to migrate in assets.
+     */
+    function migrateSDAIAssetsToNST(address receiver, uint256 assetsIn) external {
+        sdai.withdraw(assetsIn, address(this), msg.sender);
+        _migrateDAIToNST(receiver, assetsIn);
+    }
+
+    /**
+     * @notice Migrate `sharesIn` of `sdai` to `nst`.
+     * @param  receiver  The receiver of the `nst`.
+     * @param  sharesIn  The amount of the `sdai` to migrate in shares.
+     * @return assetsOut The amount of `nst` assets received.
+     */
+    function migrateSDAISharesToNST(address receiver, uint256 sharesIn) external returns (uint256 assetsOut) {
+        assetsOut = sdai.redeem(sharesIn, address(this), msg.sender);
+        _migrateDAIToNST(receiver, assetsOut);
+    }
+
+    /**
+     * @notice Migrate `assetsIn` of `sdai` to `snst`.
+     * @param  receiver  The receiver of the `snst`.
+     * @param  assetsIn  The amount of the `sdai` to migrate in assets.
+     * @return sharesOut The amount of `snst` shares received.
+     */
+    function migrateSDAIAssetsToSNST(address receiver, uint256 assetsIn) external returns (uint256 sharesOut) {
+        sdai.withdraw(assetsIn, address(this), msg.sender);
+        _migrateDAIToNST(address(this), assetsIn);
+        sharesOut = snst.deposit(assetsIn, receiver);
+    }
+
+    /**
+     * @notice Migrate `sharesIn` of `sdai` to `snst`.
+     * @param  receiver  The receiver of the `snst`.
+     * @param  sharesIn  The amount of the `sdai` to migrate in shares.
+     * @return sharesOut The amount of `snst` shares received.
+     */
+    function migrateSDAISharesToSNST(address receiver, uint256 sharesIn) external returns (uint256 sharesOut) {
+        uint256 assets = sdai.redeem(sharesIn, address(this), msg.sender);
+        _migrateDAIToNST(address(this), assets);
+        sharesOut = snst.deposit(assets, receiver);
+    }
+
+    /**
+     * @notice Downgrade `assetsIn` of `nst` to `dai`.
+     * @param  receiver The receiver of the `dai`.
+     * @param  assetsIn The amount of the `nst` to downgrade.
+     */
+    function downgradeNSTToDAI(address receiver, uint256 assetsIn) external {
+        nst.transferFrom(msg.sender, address(this), assetsIn);
+        nstJoin.join(address(this), assetsIn);
+        daiJoin.exit(receiver, assetsIn);
```

Comment:
This should be aligned the same way as in `_migrateDAIToNST`
```suggestion
        nstJoin.join(address(this), assetsIn);
        daiJoin.exit(receiver,      assetsIn);
```

Reply:
fixed